azerothcore/azerothcore-wotlk
# Core/Players: Update the player's zone and area only after update the player's position server-side

## Changes Proposed:
Port player zone and area update logic from TC.
I was looking into player map explore code and found this commit while comparing with TC. 

Sadly this does not fix the map explore issue after login mentioned in the comment of https://github.com/azerothcore/azerothcore-wotlk/issues/7737

## Issues Addressed:
- 

## SOURCE:
https://github.com/TrinityCore/TrinityCore/pull/12171

## Tests Performed:
builds, checked ingame

## How to Test the Changes:
1. create new character
2. enter new areas and zone
3. check map explore, channel messages, unmount in buildings, ...

## Known Issues and TODO List:
- 

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
